### ISSUE TITLE
[fail] multiple test failures in MSSQLServerSlickOffsetStoreSpec

### ISSUE THREAD

**Issue by ignasi35** [akka/akka-projection#322]
This is potentially related to #318 or #321.

I'm not sure what the actual cause is but it looks like all tests on `MSSQLServerSlickOffsetStoreSpec` (starting in https://travis-ci.com/github/akka/akka-projection/jobs/358713407#L585-L586) failed with similar symptoms:

```
<-- [akka.projection.slick.MSSQLServerSlickOffsetStoreSpec: The SlickOffsetStore [MS SQL Server Database] must create and update offsets] End of log messages of test that [Failed(org.scalatest.exceptions.TestFailedException: check - read offset A timeout occurred waiting for a future to complete. Queried 31 times, sleeping 100 milliseconds between each query.)]
[info] - must create and update offsets *** FAILED *** (7 seconds, 95 milliseconds)
[info]   check - read offset A timeout occurred waiting for a future to complete. Queried 31 times, sleeping 100 milliseconds between each query. (SlickOffsetStoreSpec.scala:227)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.concurrent.Futures$FutureConcept.tryTryAgain$1(Futures.scala:538)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValueImpl(Futures.scala:550)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValueImpl$(Futures.scala:479)
[info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValueImpl(ScalaFutures.scala:275)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValue(Futures.scala:476)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValue$(Futures.scala:475)
[info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValue(ScalaFutures.scala:275)
[info]   at akka.projection.slick.SlickOffsetStoreSpec.$anonfun$new$4(SlickOffsetStoreSpec.scala:227)
[info]   at org.scalatest.Assertions.withClue(Assertions.scala:1065)
[info]   at org.scalatest.Assertions.withClue$(Assertions.scala:1052)
[info]   at akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit.withClue(ScalaTestWithActorTestKit.scala:31)
[info]   at akka.projection.slick.SlickOffsetStoreSpec.$anonfun$new$2(SlickOffsetStoreSpec.scala:226)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
[info]   at org.scalatest.OutcomeOf.outcomeOf$(OutcomeOf.scala:83)
```

or

```
<-- [akka.projection.slick.MSSQLServerSlickOffsetStoreSpec: The SlickOffsetStore [MS SQL Server Database] must save and retrieve offsets of type Long] End of log messages of test that [Failed(org.scalatest.exceptions.TestFailedException: check - save offset A timeout occurred waiting for a future to complete. Queried 31 times, sleeping 100 milliseconds between each query.)]
[info] - must save and retrieve offsets of type Long *** FAILED *** (3 seconds, 34 milliseconds)
[info]   check - save offset A timeout occurred waiting for a future to complete. Queried 31 times, sleeping 100 milliseconds between each query. (SlickOffsetStoreSpec.scala:247)
[info]   org.scalatest.exceptions.TestFailedException:
[info]   at org.scalatest.concurrent.Futures$FutureConcept.tryTryAgain$1(Futures.scala:538)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValueImpl(Futures.scala:550)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValueImpl$(Futures.scala:479)
[info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValueImpl(ScalaFutures.scala:275)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValue(Futures.scala:476)
[info]   at org.scalatest.concurrent.Futures$FutureConcept.futureValue$(Futures.scala:475)
[info]   at org.scalatest.concurrent.ScalaFutures$$anon$1.futureValue(ScalaFutures.scala:275)
[info]   at akka.projection.slick.SlickOffsetStoreSpec.$anonfun$new$8(SlickOffsetStoreSpec.scala:247)
[info]   at org.scalatest.Assertions.withClue(Assertions.scala:1065)
[info]   at org.scalatest.Assertions.withClue$(Assertions.scala:1052)
[info]   at akka.actor.testkit.typed.scaladsl.ScalaTestWithActorTestKit.withClue(ScalaTestWithActorTestKit.scala:31)
[info]   at akka.projection.slick.SlickOffsetStoreSpec.$anonfun$new$7(SlickOffsetStoreSpec.scala:247)
[info]   at org.scalatest.OutcomeOf.outcomeOf(OutcomeOf.scala:85)
```

**Comment by ignasi35** [comment]
Running locally I can reproduce the issue sometimes. 

I tried running the tests disabling `parallelExecution` and noticed:

```
Jul 08, 2020 12:35:35 PM com.microsoft.sqlserver.jdbc.SQLServerConnection Prelogin
WARNING: ConnectionID:1 ClientConnectionId: 29c81c95-b8b0-4cb2-8bbf-02788bcd294f Prelogin error: host localhost port 33102 Unexpected end of prelogin response after 0 bytes read
Jul 08, 2020 12:35:35 PM com.microsoft.sqlserver.jdbc.SQLServerConnection Prelogin
WARNING: ConnectionID:1 ClientConnectionId: bd585a07-fdef-42c1-9603-bf11db2e5fc6 Prelogin error: host localhost port 33102 Unexpected end of prelogin response after 0 bytes read
Jul 08, 2020 12:35:35 PM com.microsoft.sqlserver.jdbc.SQLServerConnection Prelogin
WARNING: ConnectionID:1 ClientConnectionId: 0ae98b00-1dfc-44c6-ae91-a91673f3cc51 Prelogin error: host localhost port 33102 Unexpected end of prelogin response after 0 bytes read
Jul 08, 2020 12:35:35 PM com.microsoft.sqlserver.jdbc.SQLServerConnection Prelogin
WARNING: ConnectionID:1 ClientConnectionId: 612a7c5e-6d72-4f73-a2b5-8b86f7546a2f Prelogin error: host localhost port 33102 Unexpected end of prelogin response after 0 bytes read
Jul 08, 2020 12:35:36 PM com.microsoft.sqlserver.jdbc.SQLServerConnection Prelogin
WARNING: ConnectionID:1 ClientConnectionId: eda67d98-b81d-4048-97cf-04b3c99f454a Prelogin error: host localhost port 33102 Unexpected end of prelogin response after 0 bytes read
Jul 08, 2020 12:35:37 PM com.microsoft.sqlserver.jdbc.SQLServerConnection Prelogin
WARNING: ConnectionID:1 ClientConnectionId: b9171dd3-5af4-423c-8e4b-619fbcfb54af Prelogin error: host localhost port 33102 Unexpected end of prelogin response after 0 bytes read
Jul 08, 2020 12:35:38 PM com.microsoft.sqlserver.jdbc.SQLServerConnection Prelogin
WARNING: ConnectionID:1 ClientConnectionId: 0b3e0da6-c50c-41c1-b775-81de195f5665 Prelogin error: host localhost port 33102 Unexpected end of prelogin response after 0 bytes read
[info] MSSQLServerSlickOffsetStoreSpec:
[info] The SlickOffsetStore [MS SQL Server Database]
[info] - must create and update offsets (630 milliseconds)
...
```

When running the tests in parallel, the warning also appears, and I think it appears more times. 

I suspect, MSSQL Server takes longer to be available than other DBs. We may need a canary operation blocking the test execution until the db is 100% available.

**Comment by seglo** [comment]
This was addressed in #380
